### DISCUSSION TITLE
Difference between OpensslClientConfigBuilder and OpensslInterceptionClientConfigBuilder

### DISCUSSION THREAD

**Discussion by msolnik** [bytedance/g3#208] [Q&A] [answered]
Hello @zh-jq 
I noticed your new additions to OpensslClientConfigBuilder - such as the ability to specify the type of ciphers etc. I was curious what the difference between OpensslClientConfigBuilder and OpensslInterceptionClientConfigBuilder was for? The features you added would be great for intercepted clients too. Or do they overlap?

Thanks

**Comment by zh-jq-b** [comment, marked as answer]
OpensslClientConfigBuilder is a more general config, it will covers OpensslInterceptionClientConfigBuilder.
OpensslInterceptionClientConfigBuilder is only used in tls interception, it has fewer config options as the missing ones can be fetched from the client hello message.

**Comment by msolnik** [comment]
Great! I will use that from now own. Thanks @zh-jq-b
